### PR TITLE
Moving to latest SDK versions + adding HTTP binding tests

### PR DESCRIPTION
**Pull request by mathewc**
Pulling in the new binding path expression support and adding more e2e tests.

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/TestScripts/Node/HttpTriggerToBlob/function.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "type": "httpTrigger",
+            "name": "req",
+            "direction": "in",
+            "methods": [ "post" ]
+        },
+        {
+            "type": "blob",
+            "name": "outBlob",
+            "direction": "out",
+            "path": "test-output-node/{headers.prefix}-{id}-{query.suffix}"
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/TestScripts/Node/HttpTriggerToBlob/function.json]: This is the main test

File: src/WebJobs.Script.WebHost/Web.config [hunk 2]
```diff
@@ -202,7 +202,7 @@
       <dependentAssembly>
         <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
         <!--Explicitly redirecting back to 4.0.0. See https://github.com/dotnet/corefx/issues/16805 for details-->
-        <bindingRedirect oldVersion="0.0.0.0-4.1.1.0" newVersion="4.0.0.0" />
+        <bindingRedirect oldVersion="0.0.0.0-4.1.1.0" newVersion="4.1.1.0" />
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Web.config]: Revert this and the `System.IO.Compression` redirects (I know... painful)

Comment (mathewc) [inline src/WebJobs.Script.WebHost/Web.config]: Reverted this file, thanks

**Comment by mathewc** [comment]
VS was showing me lint warnings, so I fixed them.

**Pull request merged**